src/gui/transferlistdelegate.cpp

```diff
@@ -41,38 +41,36 @@
 #include "transferlistmodel.h"
 
 TransferListDelegate::TransferListDelegate(QObject *parent)
-    : QItemDelegate(parent)
+    : QStyledItemDelegate {parent}
 {
 }
 
 void TransferListDelegate::paint(QPainter *painter, const QStyleOptionViewItem &option, const QModelIndex &index) const
 {
     if (index.column() != TransferListModel::TR_PROGRESS) {
-        QItemDelegate::paint(painter, option, index);
+        QStyledItemDelegate::paint(painter, option, index);
         return;
     }
 
-    painter->save();
-
-    QStyleOptionViewItem opt = QItemDelegate::setOptions(index, option);
-    QItemDelegate::drawBackground(painter, opt, index);
-
     QStyleOptionProgressBar newopt;
-    newopt.rect = opt.rect;
+    newopt.rect = option.rect;
     newopt.text = index.data().toString();
-    newopt.progress = index.data(TransferListModel::UnderlyingDataRole).toInt();
+    newopt.progress = static_cast<int>(index.data(TransferListModel::UnderlyingDataRole).toReal());
```

Comment:
This change surely belongs to the previous commit.

Reply:
Moved there.